mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ConfigRefDeclarationEnricher.java

```diff
@@ -60,6 +60,14 @@ public class ConfigRefDeclarationEnricher implements DeclarationEnricher {
   private static final String CONFIG_REF_NAME = "config-ref";
   private static final MetadataType CONFIG_TYPE = buildConfigRefType();
 
+  /**
+   * @return The {@link MetadataType} suitable for a config-ref parameter.
+   * @since 1.5.0
+   */
+  public static MetadataType getConfigType() {
```

Comment:
this is for whom to consume? Not cool to have other components coupled to an enricher. If this is relevant, it should be in some constants class.  Also notice this is class is not API

Reply:
Noted, we can refactor that.
For now it was just part of the experiment of making parameters of reusable operations that declare a configuration type in the DSL, be declared with the same `MetadataType` used for config-ref parameters. I'm consuming it from the [MuleSdkParameterParameterModelParserSdk](https://github.com/mulesoft/mule/pull/11638/files#diff-d78636ed7912dc60ef62df5315b9e7bad8e0ea3fb3636681b9964ae20ce7c65aR123).

https://github.com/mulesoft/mule/pull/11638

Reply:
This should move to either ExtensionMetadataTypeUtils or ExtensionConstants. Actually, why put it in extensions-api and not keep it as an impl detail in extensions-support?

Reply:
I think we can't put it in `extension-support` unless we are also willing to move the `ConfigRefDeclarationEnricher` over there as well.
So I'm proposing to put it in an **internal** constants class in `mule-extensions-api`:
`org.mule.runtime.extension.internal.declaration.type.MetadataTypeConstants`

Please check if the package makes sense. I was inspired on these things: [DynamicConfigExpirationTypeBuilder.java](https://github.com/mulesoft/mule-extensions-api/blob/84712b1dbc09ed0df586fc6f0dd43088a7985b82/mule-extensions-api/src/main/java/org/mule/runtime/extension/api/declaration/type/DynamicConfigExpirationTypeBuilder.java#L36). 
Only for this one I guess it doesn't need to be a builder, just a constants class should be fine, and we could use it for other things also. 

Also remember this PR only makes sense together with [mule#11638](https://github.com/mulesoft/mule/pull/11638).